### PR TITLE
[1.3.5-prepare][cherry-pick]#4166 flink args build problem bug and support flink 1.10 or newer

### PR DESCRIPTION
**Pull request by chengshiwen**
#4166
issue #3987 #3457

And fix unit tests

### CODE REVIEW

File: dolphinscheduler-api/pom.xml [hunk 1]
```diff
@@ -171,6 +171,12 @@
     <dependency>
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-client</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.slf4j</groupId>
+          <artifactId>slf4j-log4j12</artifactId>
```
Comment (lenboo) [inline dolphinscheduler-api/pom.xml:177]: is there a place use this?

Comment (chengshiwen) [reply dolphinscheduler-api/pom.xml:177]: We use logback to output logs, but `slf4j-log4j12` conflicts with the `logback-classic` in ApiServer

```
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Users/chengshiwen/.m2/repository/ch/qos/logback/logback-classic/1.2.3/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/Users/chengshiwen/.m2/repository/org/slf4j/slf4j-log4j12/1.7.5/slf4j-log4j12-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
```

Comment (chengshiwen) [reply dolphinscheduler-api/pom.xml:177]: About bindings of slf4j, please see [http://www.slf4j.org/images/concrete-bindings.png](http://www.slf4j.org/images/concrete-bindings.png)

**Comment by chengshiwen** [comment]
The `skip test` is configured as `true`, resulting in all unit tests cannot be executed and **SonarCloud Quality Gate** is always **failed**.
I try configure `skip test` as `true`, but there are 8 unit test files which are failed to pass the test.
I have commented the unit test files in pom.xml, and maybe this problem needs another pr to fix.
@dailidong @CalvinKirs

**Review by break60 — APPROVED** [review]
front end +1

**Review by lenboo — APPROVED** [review]
+1

**Pull request merged**